pkg/dns/aws/dns.go

```diff
@@ -101,13 +101,26 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 	if len(region) == 0 {
 		return nil, fmt.Errorf("region is required")
 	}
+	awsconfig := &aws.Config{}
```

Comment:
i think this makes it seem like awsconfig for all clients, when this is for r53 client only, So let's name it so.